src/Monticello/MCScanner.class.st

```diff
@@ -27,14 +27,32 @@ MCScanner class >> scanTokens: aString [
 
 { #category : #actions }
 MCScanner >> next [
-	| c |
-	stream skipSeparators.
-	c := stream peek.
-	c = $# ifTrue: [c := stream next; peek].
-	c = $' ifTrue: [^ self nextString].
-	c = $( ifTrue: [^ self nextArray].
-	c isAlphaNumeric ifTrue: [^ self nextSymbol].
-	self error: 'Unknown token type'.	
+	"This would be slightly simpler using recursion but it risks a stack overflow 
+	 for some packages on some platforms, so we implement it using a local stack. 
+	 The performance is essentially equivalent (<5% difference)."
+
+	| stack |
+	stack := OrderedCollection with: OrderedCollection new.
```

Comment:
Why not using `Stack` as the stack?

Reply:
Because I'm used to GemStone that doesn't have it. I've submitted an update. Thanks!